src/app/pages/common/ix-forms/components/ix-file-upload/ix-file-upload.component.ts

```diff
@@ -0,0 +1,115 @@
+  templateUrl: './ix-file-upload.component.html',
+  styleUrls: ['./ix-file-upload.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class IxFileUploadComponent implements ControlValueAccessor {
+  @ViewChild('fileInput', { static: false }) fileInput: ElementRef<HTMLInputElement>;
+  @Input() label?: string;
+  @Input() tooltip?: string;
+  @Input() fileLocation?: string;
+  @Input() acceptedFiles?: string;
+  @Input() multiple?: boolean;
+  @Input() required?: boolean;
+  @Input() hideButton?: boolean;
+
+  fileList: FileList;
+  fbrowser: HTMLInputElement;
+
+  apiEndPoint = '/_upload?auth_token=' + this.ws.token;
+  isDisabled = true;
+
+  formControl = new FormControl(this).value as FormControl;
+
+  onChange: (value: FileList | null) => void = (): void => {};
+  onTouch: () => void = (): void => {};
+
+  constructor(
+    protected ws: WebSocketService,
+    protected http: HttpClient,
+    private loader: AppLoaderService,
+    public dialog: DialogService,
+    public translate: TranslateService,
+    public controlDirective: NgControl,
+    private cdr: ChangeDetectorRef,
+  ) {
+    this.controlDirective.valueAccessor = this;
+  }
+
+  fileBtnClick(): void {
+    this.fileInput.nativeElement.click();
+    this.fbrowser = document.getElementById('fb') as HTMLInputElement;
+    this.fbrowser.onchange = () => {
+      this.fileList = this.fileInput.nativeElement.files;
+      this.cdr.markForCheck();
+    };
+  }
+
+  upload(location = '/tmp/'): void {
+    const fileBrowser = this.fileInput.nativeElement;
+
+    if (fileBrowser.files && fileBrowser.files[0]) {
+      const formData: FormData = new FormData();
+      formData.append('data', JSON.stringify({
+        method: 'filesystem.put',
+        params: [location + '/' + fileBrowser.files[0].name, { mode: '493' }],
+      }));
+      formData.append('file', fileBrowser.files[0]);
+      const req = new HttpRequest('POST', this.apiEndPoint, formData, {
+        reportProgress: true,
+      });
+      this.loader.open();
+      this.http.request(req).pipe(untilDestroyed(this)).subscribe((event) => {
+        if (event.type === HttpEventType.UploadProgress) {
+          const percentDone = Math.round(100 * event.loaded / event.total);
+          const uploadMsg = `${percentDone}% ${this.translate.instant('Uploaded')}`;
+          this.loader.dialogRef.componentInstance.title = uploadMsg;
+        } else if (event instanceof HttpResponse) {
+          if (event.statusText === 'OK') {
+            this.loader.close();
+            this.dialog.info(this.translate.instant('File upload complete'), '', '300px', 'info', true);
+          }
+        }
+      }, (error) => {
+        this.loader.close();
+        this.dialog.errorReport(this.translate.instant('Error'), error.statusText, error.message);
+      });
+    } else {
+      this.dialog.info(this.translate.instant('Please make sure to select a file'), '', '300px', 'info', true);
+    }
+  }
```

Comment:
This breaks our logic of having dumb components that don't make requests on their own.
This will create problems in tests and will make it impossible to customize component behavior elsewhere.
For example, here are some other examples, where people were reinventing the wheel, because they needed to upload to a different API endpoint.
<img width="965" alt="Снимок экрана 2021-12-06 в 13 23 30" src="https://user-images.githubusercontent.com/1481270/144837854-6dff9fc2-6af5-46c3-b680-a43a1d446696.png">

Ideally we want components to only have presentation logic and extract actual network logic somewhere else. See what we did for `IxFormatterService` and `IxInputComponent.format`. Maybe we could do something similar.
Alternatively we could extract uploading code in a service and consume this service in a component that hosts ix-file-upload.
On this note, it's probably a good idea to call this `ix-file-input`, not `ix-file-upload`.

Reply:
Also not resolved. The point here is not add an option to let developer override default behavior. The point is to keep the component dumb and not have network requests in it at all. Network requests need to happen outside of the file input component in a separate reusable code.
Could be a service method that you as a developer connect in a parent component when building a form. 
We can also explore the possibility that file-input doesn't do any of the uploading on file select, but it's the host form that uploads a file when the whole form is submitted.

I would also suggest you pick any new form and add a file input there, so that it's easier to see how it works.